File: pdv-desktop/src/renderer/src/lib/currency.ts
// Mesma convenção do backend Armazix: preço trafega e é armazenado como
// string decimal ("12.90"), nunca float — evita erro de arredondamento em
// dinheiro. Só vira number no último passo, pra exibir.
export function formatPrice(value: string | number): string {
  const n = typeof value === "string" ? parseFloat(value) : value;
  return (Number.isFinite(n) ? n : 0).toLocaleString("pt-BR", {
    minimumFractionDigits: 2,
    maximumFractionDigits: 2,
  });
}

export function money(n: number): string {
  return (Math.round(n * 100) / 100).toFixed(2);
}

File: pdv-desktop/lib/core/utils/currency.dart
import 'package:intl/intl.dart';

/// Preço trafega e é armazenado como string decimal ("12.90"), nunca double
/// puro — mesma convenção do backend Armazix, evita erro de arredondamento
/// em dinheiro. Só vira double no último passo, pra formatar.
class Currency {
  Currency._();

  static final _formatter = NumberFormat.currency(locale: 'pt_BR', symbol: 'R\$');

  static double parse(String value) => double.tryParse(value) ?? 0;

  static String format(num value) => _formatter.format(value);

  static String formatFromString(String value) => format(parse(value));

  static String toApiString(double value) => value.toStringAsFixed(2);
}
